File: Runtime/Marvin.Runtime.Maintenance/Plugins/ModuleMaintenance/Wcf/Models/ModuleServerModuleState.cs
﻿namespace Marvin.Runtime.Maintenance.Plugins.ModuleMaintenance.Wcf.Models
{
    /// <summary>
    /// Server module state
    /// </summary>
    public enum ModuleServerModuleState
    {
        /// <summary>
        /// Initial value
        /// </summary>
        Stopped,

        /// <summary>
        /// Module is initializing
        /// </summary>
        Initializing,

        /// <summary>
        /// Service is ready to be started
        /// </summary>
        Ready,

        /// <summary>
        /// Service is starting
        /// </summary>
        Starting,

        /// <summary>
        /// Service is running
        /// </summary>
        Running,

        /// <summary>
        /// Service is stopping
        /// </summary>
        Stopping,

        /// <summary>
        /// Service failed
        /// </summary>
        Failure,
    }
}

File: src/Moryx.CommandCenter.Web/app/src/modules/models/ModuleServerModuleState.ts
/*
 * Copyright (c) 2026 Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

export enum ModuleServerModuleState {
    Stopped = "Stopped",
    Initializing = "Initializing",
    Ready = "Ready",
    Starting = "Starting",
    Running = "Running",
    Stopping = "Stopping",
    Failure = "Failure",
    Missing = "Missing",
}
